src/com/maddyhome/idea/vim/ui/ExTextField.java

```diff
@@ -359,17 +361,14 @@ private void resetCaret() {
   // see :help 'guicursor'
   // Note that we can't easily support guicursor because we don't have arbitrary control over the IntelliJ editor caret
   private void setNormalModeCaret() {
-    CommandLineCaret caret = (CommandLineCaret) getCaret();
     caret.setBlockMode();
```

Comment:
is it any possibilities that caret would be invalidated and `caret.isValid()` would be `false`?